addons/binding/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/handler/EvohomeHeatingZoneHandler.java

```diff
@@ -14,30 +14,41 @@
 import org.eclipse.smarthome.core.thing.Thing;
 import org.eclipse.smarthome.core.thing.ThingStatus;
 import org.eclipse.smarthome.core.thing.ThingStatusDetail;
-import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
 import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
 import org.openhab.binding.evohome.EvohomeBindingConstants;
 import org.openhab.binding.evohome.internal.api.EvohomeApiClient;
 import org.openhab.binding.evohome.internal.api.models.v2.response.ZoneStatus;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * The {@link EvohomeHandler} is responsible for handling commands, which are
+ * The {@link EvohomeHeatingZoneHandler} is responsible for handling commands, which are
  * sent to one of the channels.
  *
  * @author Jasper van Zuijlen - Initial contribution
  */
-public class EvohomeHandler extends BaseThingHandler {
+public class EvohomeHeatingZoneHandler extends BaseEvohomeHandler {
 
-    private final Logger logger = LoggerFactory.getLogger(EvohomeHandler.class);
+    private final Logger logger = LoggerFactory.getLogger(EvohomeHeatingZoneHandler.class);
 
-    public EvohomeHandler(Thing thing) {
+    private int zoneId;
+
+    public EvohomeHeatingZoneHandler(Thing thing) {
         super(thing);
     }
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
+        if(command instanceof RefreshType){
+            if(getBridge() != null && getBridge().getHandler() != null){
+                EvohomeApiClient apiClient = ((EvohomeGatewayHandler) getBridge().getHandler()).getApiClient();
+                int zoneId = Integer.valueOf(getThing().getProperties().get(EvohomeBindingConstants.ZONE_ID));
+                int locationId = Integer.valueOf(getThing().getProperties().get(EvohomeBindingConstants.LOCATION_ID));
+                ZoneStatus zoneStatus = apiClient.getHeatingZone(locationId, zoneId);
```

Comment:
I think you are safe to assume that a `zone id` is unique, so `locationId` isn't strictly necessary. That said, I does help the search routine.